### PR TITLE
Remove http-server admin and jmx-http-rpc

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by wendigo** [comment]
the only reference to `jmx-http-rpc` that I found was an exclude in Trino :)

**Comment by wendigo** [comment]
@electrum @dain PTAL

**Comment by wendigo** [comment]
@electrum @dain PTAL

**Comment by wendigo** [comment]
@electrum PTAL

**Comment by dain** [comment]
I'm cool with dropping these

**Comment by wendigo** [comment]
@electrum please review

File: http-server/src/main/java/io/airlift/http/server/HttpServerProvider.java [hunk 1]
```diff
@@ -70,9 +67,9 @@ public HttpServerProvider(HttpServerInfo httpServerInfo,
             NodeInfo nodeInfo,
             HttpServerConfig config,
             Optional<HttpsConfig> httpsConfig,
-            @TheServlet Servlet theServlet,
-            @TheServlet Set<Filter> filters,
-            @TheServlet Set<HttpResourceBinding> resources,
+            Servlet theServlet,
+            Set<Filter> filters,
+            Set<HttpResourceBinding> resources,
```
Comment (electrum) [inline http-server/src/main/java/io/airlift/http/server/HttpServerProvider.java:72]: It feels a bit strange to inject these `jakarta.servlet` directly without an annotation, but we own the injector and I can't think of any reason why application code would inject these, so I guess it's fine.

**Review by electrum — APPROVED** [review]
Nice! These are good cleanups.

Can you split the removal of `jmx-http-rpc` into a separate commit? It seems unrelated to the admin server.

**Comment by electrum** [comment]
Future cleanup ideas:

* Remove `LoginService` binding?
* Remove `TraceTokenManager` in `HttpServerProvider` and instead bind `TraceTokenFilter` via Guice

File: http-server/src/main/java/io/airlift/http/server/HttpServerProvider.java (hunk 1 above)
Comment (wendigo) [reply http-server/src/main/java/io/airlift/http/server/HttpServerProvider.java:72]: Actually it's a preparation to reintroduce annotated Servlets, Filters along with the binder for HttpServer :)

**Pull request merged**